packages/x-date-pickers/src/PickersCalendarHeader/PickersCalendarHeader.tsx

```diff
@@ -287,6 +287,8 @@ const PickersCalendarHeader = React.forwardRef(function PickersCalendarHeader<TD
     return null;
   }
 
+  const label = `${utils.format(month, 'month')} ${utils.format(month, 'year')}`;
```

Comment:
I fear that this change might cause some backlash from users already accustomed to overriding the header label by providing the `montAndYear` format. 🤔 
https://github.com/mui/mui-x/issues/4463#issuecomment-1096464922

Correct me if I'm wrong, but after this change, there would no longer be an option to do that.
Shouldn't we still provide an easy option to achieve it in one way or another? 🤔 
Maybe a `headerLabelFormat` prop on the `calendarHeader` slot? 🤔

Reply:
Yes it's the main trade-off indeed
The new prop can be a good workaround.
Or we allow people to override our `resolveDateFormat` method (same for time and date-time) and we use it also for the header.
That way everything that we doable before is still doable but it also covers all the new views combination correctly.

Reply:
That also sounds like an interesting compromise.
However, it could possibly be more difficult and verbose if you only want to change the behavior of the `calendarHeader` label. 🤷 🤔

Reply:
Yes, it's a lot more versatile but a lot less direct to implement.

If we feel like there is not many places where it's actually useful to override those concatenated formats, then I do agree that having 2-3 props would be simpler

Reply:
Gotcha.
In both cases, those changes do not seem like breaking ones, hence, I'm not pushing for providing a solution now.
I'd love to avoid the need for props that are not necessary, hence, we might wait for someone actually having a need for it to actually add this behavior. 👍